src/coreclr/System.Private.CoreLib/src/System/ArgIterator.cs

```diff
@@ -6,128 +6,148 @@
 
 namespace System
 {
-    // This class will not be marked serializable
     // Note: This type must have the same layout as the CLR's VARARGS type in CLRVarArgs.h.
     // It also contains an inline SigPointer data structure - must keep those fields in sync.
     [StructLayout(LayoutKind.Sequential)]
-    public ref struct ArgIterator
+    public unsafe ref partial struct ArgIterator
     {
-        private IntPtr ArgCookie;               // Cookie from the EE.
+        private IntPtr _argCookie;              // Cookie from the EE.
 
-        // The SigPointer structure consists of the following members.  (Note: this is an inline native SigPointer data type)
-        private IntPtr sigPtr;                  // Pointer to remaining signature.
-        private IntPtr sigPtrLen;               // Remaining length of the pointer
+        // Note: this is an inline native SigPointer data type.
+        private IntPtr _sigPtr;                 // Pointer to remaining signature.
 
-        // Note, sigPtrLen is actually a DWORD, but on 64bit systems this structure becomes
+        // Note, this is actually a UInt32, but on 64bit systems SigPointer structure becomes
```

Comment:
Could we spilt out the inline `SigPointer` type to a private nested struct and have a field of it in `ArgIterator` to get the alignment automatically instead of the manual adjustments and various comments around it here?

Reply:
Good idea. Fixed.